controllers/virtualmachinepublishrequest/virtualmachinepublishrequest_controller.go

```diff
@@ -900,15 +916,17 @@ func (r *Reconciler) ReconcileNormal(ctx *context.VirtualMachinePublishRequestCo
 		if deleted {
 			skipPatch = true
 		}
-		isDeleted = deleted
-		return ctrl.Result{RequeueAfter: requeueAfter}, err
+		return ctrl.Result{RequeueAfter: requeueAfter}, false, err
 	}
 
-	return requeueResult(ctx), nil
+	return requeueResult(ctx), false, nil
 }
 
 func (r *Reconciler) ReconcileDelete(ctx *context.VirtualMachinePublishRequestContext) (ctrl.Result, error) {
-	r.Metrics.DeleteMetrics(ctx.Logger, ctx.VMPublishRequest.Name, ctx.VMPublishRequest.Namespace)
-	// no op
+	if controllerutil.ContainsFinalizer(ctx.VMPublishRequest, finalizerName) {
```

Comment:
Should we also return a `skipPatch` from this function to avoid patching when the resource doesn't contain the finalizer?

Reply:
It will be a noop, or even the patch failed, it is no harm. I'll leave it as it is. `SkipPatch` is needed due to CL API limitations. We have to ensure that each attempt, we only trigger one `CreateOVF` API, the .status().Update() will deny requests if the resourceVersion doesn't match.